src/Microsoft.Extensions.FileProviders.Physical/PollingFileChangeToken.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Extensions.FileProviders.Physical
+{
+    public class PollingFileChangeToken : IChangeToken
+    {
+        private FileInfo _fileInfo;
+        private DateTime _previousWriteTimeUtc;
+        private DateTime _lastCheckedTimeUtc;
+
+        public PollingFileChangeToken(FileInfo fileInfo)
+        {
+            _fileInfo = fileInfo;
+            _previousWriteTimeUtc = GetLastWriteTimeUtc();
+        }
+
+        // Internal for unit testing
+        internal static TimeSpan PollingInterval { get; set; } = TimeSpan.FromSeconds(4);
+
+        private DateTime GetLastWriteTimeUtc()
+        {
+            _fileInfo.Refresh();
+            return _fileInfo.Exists ? _fileInfo.LastWriteTimeUtc : DateTime.MinValue;
+        }
+
+        public bool ActiveChangeCallbacks => false;
+
+        public bool HasChanged
```

Comment:
Removed the need for locking. We don't cache these tokens any more so we don't have to concern ourselves with concurrency